### PR TITLE
refactor: enable active global environment and map it to the sdk

### PR DESCRIPTION
**Pull request by ihexxa**
### Background
This changes hooks the active global environment to the script runtime, and enable them in scripting.
Ref: INS-3988
Tests are failing because it requires #7511

### Changes
- [x] refactor: enable active global environment and map it to the sdk
- [x] send global env to the script runtime
- [x] persist global env
- [x] test it

### CODE REVIEW

File: packages/insomnia/src/ui/routes/request.tsx [hunk 1]
```diff
@@ -423,21 +423,25 @@ export const sendAction: ActionFunction = async ({ request, params }) => {
...
+        globals,
       });
       window.main.completeExecutionStep({ requestId });
       if (!postMutatedContext?.request) {
         // exiy early if there was a problem with the pre-request script
         // TODO: improve error message?
         return null;
       }
-      await savePatchesMadeByScript(postMutatedContext, requestData.environment, baseEnvironment);
+      await savePatchesMadeByScript(postMutatedContext, requestData.environment, baseEnvironment, globals);
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/request.tsx:444]: lets test this a bunch of times and document the various cases in the appropriate place.

File: packages/insomnia-sdk/src/objects/insomnia.ts [hunk 2]
```diff
@@ -121,13 +116,19 @@ export async function initInsomniaObject(
...
-    const baseEnvironment = new Environment(rawObj.baseEnvironmentName || '', rawObj.baseEnvironment);
+    const globals = new Environment('globals', rawObj.globals || {}); // could be undefined
+    const baseEnvironment = new Environment(rawObj.baseEnvironment.name || '', rawObj.baseEnvironment.data);
+    // reuse baseEnvironment when the "selected envrionment" points to the base environment
+    const environment = rawObj.baseEnvironment.id === rawObj.environment.id ?
+        baseEnvironment :
+        new Environment(rawObj.environment.name || '', rawObj.environment.data);
+    if (rawObj.baseEnvironment.id === rawObj.environment.id) {
+        log('warning: No environment is selected, modification of insomnia.environment will be applied to the base environment.');
+    }
```
Comment (jackkav) [inline packages/insomnia-sdk/src/objects/insomnia.ts]: same as other comment, lets document these cases

File: packages/insomnia-sdk/src/objects/insomnia.ts [hunk 3]
```diff
@@ -121,13 +116,19 @@ export async function initInsomniaObject(
     rawObj: RequestContext,
     log: (...args: any[]) => void,
 ) {
```
Comment (ihexxa) [inline packages/insomnia-sdk/src/objects/insomnia.ts:118]: nit: add some comments here for the potential mapping strategy in the future. cc @gatzjames

File: packages/insomnia/src/ui/routes/request.tsx (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/routes/request.tsx:444]: Tested most of cases such as reading, writing and so on.

File: packages/insomnia-sdk/src/objects/insomnia.ts (hunk 2 above)
Comment (ihexxa) [reply packages/insomnia-sdk/src/objects/insomnia.ts]: Added some comments for elaborating mapping rules.

**Comment by jackkav** [comment]
```
insomnia.globals.set('junk','00')
// insomnia.environment.set('junk','00')


console.log('g'+insomnia.globals.get('junk'))
console.log('e'+insomnia.environment.get('junk'))
console.log('cv'+insomnia.collectionVariables.get('junk'));
console.log('v'+insomnia.variables.get('junk'));

insomnia.globals.set('junk','000')
insomnia.collectionVariables.set('junk','44')
insomnia.environment.set('junk','what?')
// inconsistency
// currently updating selected (exclusive of globals) (inclusive of base env(collectionVariables))
// pm.env = selected env (exclusive of base env)

console.log('g'+insomnia.globals.get('junk'))
console.log('e'+insomnia.environment.get('junk'))
console.log('cv'+insomnia.collectionVariables.get('junk'));
console.log('v'+insomnia.variables.get('junk'));
```

**Review by CurryYangxx — APPROVED** [review]

**Pull request merged**